crates/core/src/factories/core_space.rs

```diff
@@ -92,13 +134,142 @@ impl Space for CoreSpace {
 
     fn local_agent_join(
         &self,
-        _local_agent: agent::DynLocalAgent,
+        local_agent: agent::DynLocalAgent,
     ) -> BoxFut<'_, K2Result<()>> {
-        Box::pin(async move { todo!() })
+        Box::pin(async move {
+            // set some starting values
+            local_agent.set_tgt_storage_arc_hint(DhtArc::Empty);
+            local_agent.set_cur_storage_arc(DhtArc::Empty);
+
+            // update our local map
+            self.inner
+                .lock()
+                .unwrap()
+                .local_agent_map
+                .insert(local_agent.agent().clone(), local_agent.clone());
+
+            // TODO - inform gossip module of new join
+            // TODO - inform sharding module of new join
+
+            let inner = self.inner.clone();
+            let space = self.space.clone();
+            let local_agent2 = local_agent.clone();
+            let peer_store = self.peer_store.clone();
+            local_agent.register_cb(Arc::new(move || {
+                let inner = inner.clone();
+                let space = space.clone();
+                let local_agent2 = local_agent2.clone();
+                let peer_store = peer_store.clone();
+                tokio::task::spawn(async move {
+                    // TODO - call an update function on the gossip module.
+                    // TODO - call an update function on the sharding module.
+
+                    let url = inner.lock().unwrap().current_url.clone();
+
+                    if let Some(url) = url {
+                        // sign a new agent info
+                        let created_at = Timestamp::now();
+                        let expires_at = created_at
+                            + std::time::Duration::from_secs(60 * 20);
+                        let info = agent::AgentInfo {
+                            agent: local_agent2.agent().clone(),
+                            space,
+                            created_at,
+                            expires_at,
+                            is_tombstone: false,
+                            url: Some(url),
+                            storage_arc: local_agent2.get_cur_storage_arc(),
```

Comment:
The target storage arc is only available on local for now which I think makes sense. The host needs to have some influence on that value and Kitsune needs to be able to adjust it. Is it not useful to other peers though? I guess they can't take it into account for arc sizing calculations because that would be wrong. I do think it's wanted in gossip though, I'm still puzzling that out. So for gossip maybe we need to be sending that information as part of sharing agent info

Reply:
For sure gossip needs to be dealing with the arcs as we're trying to sync them, not the currently signed agent storage arcs. I don't believe the targets belong in the signed agent infos, but rather should be communicated directly in the gossip messages.

Reply:
Sweet, that lines up nicely with my thinking out loud :)